components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,13 +164,65 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.home(ctx, np); err != nil {
+	if err := oAx.home(context.Background(), np); err != nil {
 		return nil, err
 	}
 
+	oAx.checkHit(context.Background())
 	return oAx, nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	ctx, g.cancelFunc = context.WithCancel(ctx)
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return nil
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				g.mu.Lock()
+				defer g.mu.Unlock()
+				if hit0, _ := g.limitHit(ctx, true); hit0 {
```

Comment:
[nit] let's change the signatures of limitHit and moveAway to take in a limit switch pin as an input to make it more readable.

Let's also check the error?
would look something like:
```go
backwardsHit , err := g.limitHit(ctx, g.limitSwithPin[0]); 
if err != nil {
      }
backwardsHit {
        g.moveAway(ctx, revsToMoveAway)
   }
```